exercises/grains/.meta/src/reference/java/Grains.java

```diff
@@ -0,0 +1,25 @@
+import java.math.BigInteger;
+import java.util.stream.IntStream;
+
+class Grains {
+
+    static BigInteger computeNumberOfGrainsOnSquare(final int square) {
+        if (1 <= square  && square <= 64) {
+            return BigInteger.valueOf(2).pow(square - 1);
+        } else {
+            throw new IllegalArgumentException("square must be between 1 and 64");
+        }
+    }
+
+    static BigInteger computeTotalNumberOfGrainsOnBoard() {
+        return IntStream.rangeClosed(1, 64)
+                .mapToObj(Grains::computeNumberOfGrainsOnSquare)
+                .reduce(
+                        BigInteger.valueOf(0),
+                        BigInteger::add);
+        // Note that the following is also correct, using the sum of the elements of a geometric series
```

Comment:
Please remove comment. This is just the reference implementation, but it is of course not the only way a student may implement it.